### PR TITLE
memory: add test for call_and_shelve performance

### PR DESCRIPTION
**Pull request by aabadie**
This test verifies that the `call_and_shelve` function doesn't perform a load of cached data every time it is used.

The idea is to verify that the access time of the file containing the cached file. It should remain untouched after each calls. And getting result using `MemorizedResult.get` will change the access time since the data must be read.

See #757

### CODE REVIEW

**Comment by aabadie** [comment]
Ok, it's failing on windows because the access time is not accurate enough (I found [this](https://stackoverflow.com/a/943532) that explains why).
So the new test is now skipped on NT based OSes.

**Comment by aabadie** [comment]
@ogrisel, I updated the test the way we discussed IRL.

File: joblib/test/test_memory.py [hunk 1]
```diff
@@ -514,6 +514,41 @@ def test_call_and_shelve_argument_hash(tmpdir):
         in str(w[-1].message)
 
 
+def test_call_and_shelve_performance(tmpdir):
```
Comment (ogrisel) [inline joblib/test/test_memory.py]: `_performance` is really too generic a name. Please rename to "test_call_and_shelve_lazily_load_stored_result".

File: joblib/test/test_memory.py [hunk 2]
```diff
@@ -514,6 +514,41 @@ def test_call_and_shelve_argument_hash(tmpdir):
...
+    test_access_time = os.stat(test_access_time_file.strpath).st_atime
+    # check file system access time stats resolution is lower than test wait
+    # timings.
+    time.sleep(0.5)
+    assert test_access_time_file.read() == 'test_access'
+
+    if test_access_time == os.stat(test_access_time_file.strpath).st_atime:
+        # Skip this test when access time cannot be retrieved with enough
+        # precision from the file system (e.g. NTFS on windows).
+        return
```
Comment (ogrisel) [inline joblib/test/test_memory.py]: Please use `pytest.skip("filesystem does not support fine-grained access time attribute")` to make it explicitly that the test is skipped.

**Comment by aabadie** [comment]
@ogrisel, comments addressed

**Comment by aabadie** [comment]
The appveyor failure seems unrelated

**Comment by aabadie** [comment]
I'v retriggered the CI and now it's all green again.

**Pull request merged**